articles/custom_storage_location.md

```diff
@@ -254,3 +254,39 @@ By default, it uses Synapse storage.  To use the external bucket configured abov
 ____
 
 
+##Synapse Proxy for access to restricted filesystem
+
+###Connecting synapse to the restricted filesystem
+You must have a key "your_sftp_key" to allow synapse to interact with the filesystem
+```python
+import synapseclient
+import json
+syn = synapseclient.login()
+PROJECT = 'syn12345'
+destination = {"uploadType":"SFTP", 
+               "secretKey":"your_sftp_key", 
+               "proxyUrl":"https://your-proxy.prod.sagebase.org", 
+               "concreteType":"org.sagebionetworks.repo.model.project.ProxyStorageLocationSettings"}
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
+project_destination ={"concreteType": "org.sagebionetworks.repo.model.project.UploadDestinationListSetting", 
+                      "settingsType": "upload"}
+project_destination['locations'] = destination['storageLocationId']
+project_destination['projectId'] = PROJECT
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
+```
+
+####Create a fileHandle
+A filehandle is merely a synapse representation of the file, therefore you will have to specify all the metadata below for synapse to recognize it.
+```python
+path='/path/to/your/file.txt'
+fileHandle = {'concreteType': 'org.sagebionetworks.repo.model.file.ProxyFileHandle',
+              'fileName'    : os.path.split(path)[-1],
+              'contentSize' : "2252439673",
+              'filePath' : path,
+              'contentType' : "application/octet-stream",
+              'contentMd5' :  '67a3688466ad3f606e2cc7b42df4d4bb',
+              'storageLocationId': destination['storageLocationId']}
+fileHandle = syn.restPOST('/externalFileHandle/proxy', json.dumps(fileHandle), endpoint=syn.fileHandleEndpoint)
+f = synapseclient.File(parentId=PROJECT, dataFileHandleId = fileHandle['id'])
+f = syn.store(f)
+```
```

Comment:
Tom, nice code examples.  Can you fill in some of the details, as to why you would use a proxy how does a proxy work and link to the documentation in github about setting up the proxy.  You might be able to take some of the text from John's wiki in github.  That is,  assume that someone has no idea what is going on and you need to bring them up to speed to create content in Synapse.